workflow/util/pod_name.go

```diff
@@ -58,8 +58,8 @@ func GeneratePodName(workflowName, nodeName, templateName, nodeID string, versio
 	}
 
 	prefix := workflowName
-	if !strings.Contains(nodeName, ".inline") {
-		prefix = fmt.Sprintf("%s-%s", workflowName, templateName)
+	if !strings.Contains(nodeName, ".inline") && templateName != "" {
```

Comment:
```suggestion
	if templateName != "" {
```

this can be simplified and would match the UI, per my previous review comment https://github.com/argoproj/argo-workflows/pull/12928#discussion_r1575035131